mlfinlab/backtest_statistics/statistics.py

```diff
@@ -0,0 +1,233 @@
+    - entry_time = (previous_time * weight_of_previous_position + 
+                    time_since_beginning_of_trade * increase_in_position ) /
+                    weight_of_current_position
+    - holding_period ['holding_time' = time a position was held, 'weight' = weight of position closed]
+    - res = weighted average time a trade was held
+    
+    :param target_positions: (pd.Series) target position series with timestamps as indices
+    :return: (float) estimated average holding period, NaN if zero or unpredicted
+    """
+    holding_period, entry_time = pd.DataFrame(columns=['holding_time','weight']), 0
+    position_difference = target_positions.diff() 
+    
+    # Time elapsed from the starting time for each position
+    time_difference = (target_positions.index-target_positions.index[0]) / np.timedelta64(1,'D')
+    for i in range(1, target_positions.shape[0]):
+        if float(position_difference.iloc[i] * target_positions.iloc[i-1]) >= 0: # Increased or unchanged position
+            if float(target_positions.iloc[i]) != 0: # And not an empty position
+                entry_time = (entry_time * target_positions.iloc[i-1] + 
+                              time_difference[i] * position_difference.iloc[i]) / target_positions.iloc[i]
+        else: # Decreased
+            if float(target_positions.iloc[i] * target_positions.iloc[i-1]) < 0: # Flip of a position
+                holding_period.loc[target_positions.index[i], ['holding_time','weight']] = (time_difference[i] - entry_time, abs(target_positions.iloc[i-1]))
+                entry_time = time_difference[i] # Reset entry time
+            else: # Only a part of position is closed
+                holding_period.loc[target_positions.index[i], ['holding_time','weight']] = (time_difference[i] - entry_time, abs(position_difference.iloc[i]))
+                
+    if float(holding_period['weight'].sum()) > 0: # If there were closed trades at all
+        res = float((holding_period['holding_time'] * holding_period['weight']).sum() / holding_period['weight'].sum())
+    else:
+        res = float('nan')
+        
+    return res
+
+
+def bets_concentration(returns: pd.Series) -> float:
+    """
+    Snippet 14.3, page 201, Derives the concentration of bets across months
+    from given pd.Series of returns.
+    
+    Algorithm is based on Herfindahl-Hirschman Index where return weights
+    are taken as an input.
+    
+    :param returns: (pd.Series) returns from bets
+    :return: (float) concentration of returns (nan if less than 3 returns)
+    """  
+    if returns.shape[0]<=2:
+        return float('nan') # If less than 3 bets
+    weights = returns / returns.sum() # Weights of each bet
+    hhi = (weights ** 2).sum() # Herfindahl-Hirschman Index for weights
+    hhi = float((hhi - returns.shape[0] **(-1)) / (1 - returns.shape[0]**(-1)))
+    return hhi
+
+
+def positive_negative_bets_concentration(returns: pd.Series, freq: str = 'M') -> tuple:
```

Comment:
@proskurin, I have mixed feelings about using the strong typed variables here. I think it makes the documentation on ReadTheDocs less readable. I can understand why its desired from a code point of view but these types are also not enforced are they?

I'd love to hear your thoughts on this.